packages/semver/src/schematics/ng-add/index.ts

```diff
@@ -0,0 +1,48 @@
+import { chain, Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { addPackageJsonDependency, NodeDependency, NodeDependencyType } from '@schematics/angular/utility/dependencies';
+import { of } from 'rxjs';
+import { concatMap, map } from 'rxjs/operators';
+
+import { getLatestNodeVersion } from '../../utils/npm';
+import { SchemaOptions } from './schema';
+
+const PACKAGE_NAME = '@jscutlery/semver';
+
+function addDependency(): Rule {
```

Comment:
Do we really need this?

Doesn't `ng add` already add the schematic to `devDependencies`? 🤔

Reply:
Good catch, we could define it in the `package.json` as following : 

```
"ng-add": {
  "save": "devDependencies"
}
```

Reply:
If we use [the trick mentioned above](https://angular.io/guide/schematics-for-libraries#define-dependency-type) we loose the ability to test the `package.json` with the new entry `@jscutlery/semver`. Somehow the `package.json` update is made outside the testing context. 

I checked the Nx repo and they use the same technique as I did before, which can be tested but more verbose : 

```ts
function updateDependencies(): Rule {
  return updateJsonInTree('package.json', (json) => {
    delete json.dependencies['@nrwl/node'];
    json.devDependencies['@nrwl/node'] = nxVersion;
    return json;
  });
```

What do you think ?

Reply:
I don't know why they did this.
We don't have to test that the dependency is added to the package json as this is how angular handles the ng-add. We don't have to test how ng-add works.

We can add an e2e test if this has any chance to break.